src/business/form.ts

```diff
@@ -1,10 +1,16 @@
+import {FieldListenerRepository, FieldListenerRepositoryImpl, Listener} from './field_listener_repository';
 export class Form {
-
     private values: { [key: string]: any } = {};
 
     private validationErrors: { [fieldName: string]: any } = {};
 
-    private fieldListeners: { [fieldName: string]: Array<(value: any) => void> } = {};
+    private _fieldListenerRepository: FieldListenerRepository;
+
+    constructor(fieldListenerRepository: FieldListenerRepository | null = null) {
```

Comment:
why a `| null` type instead of making the `fieldListenerRepository` nullable? Aren't we forced to pass `null` in the constructor call now? (not sure about this, but still a bit confusing)